src/svelte/Components/Modals/MacroModal/MacroModal.svelte

```diff
@@ -0,0 +1,158 @@
+<script>
+  import Error from "../../Alerts/Error.svelte";
+  import InputWrapper from "../../Inputs/InputWrapper.svelte";
+  import RunInfo from "./RunInfo.svelte";
+
+  let toggle;
+  let errors = {
+    name: false,
+    command: false,
+  };
+
+  export let type;
+  export let heading;
+  export let subHeading;
+  export let submitText;
+  export let onSubmit;
+  export let values = {
+    name: "",
+    description: "",
+    command: "",
+    runnable: false,
+  };
+
+  function isUnSet(value) {
+    return value.trim() === "";
+  }
+
+  function submit() {
+    //error checker for any error
+    let err = false;
+
+    //check name
+    if (isUnSet(values.name)) {
+      errors.name = true;
+      err = true;
+    }
+
+    //check command when runnable
+    if (values.runnable && isUnSet(values.command)) {
+      errors.command = true;
+      err = true;
+    }
+
+    //exit when no error
+    if (err) return;
```

Comment:
```suggestion
    if (error.name || error.command) return;
```
wirklich so unleserlich? würde einiges an Zeilen sparen